### ISSUE TITLE
soft deprecation for jsx-space-before-closing

### ISSUE THREAD

**Issue by dnalborczyk** [yannickcr/eslint-plugin-react#1137]
The introduction of _jsx-tag-spacing_ makes _jsx-space-before-closing_ obsolete.

https://github.com/yannickcr/eslint-plugin-react/pull/1070
https://github.com/yannickcr/eslint-plugin-react/issues/693#issuecomment-234842809

since the plan is to deprecate _jsx-space-before-closing_, I think it would be already helpful and less confusing - as it was for me - if this would be already mentioned in the docs.

something like:
```
Note: This rule will be deprecated in the next major release. Consider using jsx-tag-spacing instead.
```

Mind if I do a PR?

**Comment by ljharb** [comment]
Sounds great!

You can also mark the rule itself as deprecated in its metadata.

**Comment by tleunen** [comment]
I believe this is a must have. Because if a project has both configuration, with different settings, eslint will keep complaining that there's an error.

**Comment by alexzherdev** [comment]
Looks like this is already in the docs via https://github.com/yannickcr/eslint-plugin-react/pull/1070/files#diff-682c810c832cf33406eede243921fd42

**Comment by ljharb** [comment]
Closed with #1070.
